src/minirouter/ip.go

```diff
@@ -109,72 +110,64 @@ func handleIP(c *minicli.Command, r chan<- minicli.Responses) {
 }
 
 func ipDel(idx int, ip string) error {
-	if idx >= len(IPs) {
-		return fmt.Errorf("invalid index: %v", idx)
-	}
-
-	var found bool
-	for _, v := range IPs[idx] {
-		if ip == v {
-			found = true
-			break
+	iface := "lo"
+	if idx != -1 {
+		// get an interface from the index
+		var err error
+		iface, err = findEth(idx)
+		if err != nil {
+			return err
 		}
 	}
-	if !found {
-		return fmt.Errorf("no such ip: %v", ip)
-	}
-
-	// get an interface from the index
-	eth, err := findEth(idx)
-	if err != nil {
-		return err
-	}
 
-	// TODO: what's the right way to remove a dhcp interface?
+	// TODO: Need to send release ip reservation to dhcp server
 	if ip == "dhcp" {
+		out, err := exec.Command("dhclient -r", iface).CombinedOutput()
```

Comment:
This should be `exec.Command("dhclient", "-r", iface).CombinedOutput()`

Reply:
Done